auth/src/subscription.rs

```diff
@@ -24,21 +24,28 @@ where
     async fn from_request(req: Request<B>, state: &S) -> Result<Self, Self::Rejection> {
         // Extract the NewSubscriptionItem, the struct that other services should use when calling
         // the endpoint to add subscription items.
-        let NewSubscriptionItem { quantity, item } = axum::Json::from_request(req, state)
+        let NewSubscriptionItem {
+            metadata_id,
```

Comment:
as we've discussed, I believe we can still have this feature without using metadata, given we should be able to update the quantity and not care about the underlying RDS instances (given they are the same).